### DISCUSSION TITLE
图片文件会进行数据修改？

### DISCUSSION THREAD

**Discussion by zhendery** [messense/aliyundrive-webdav#372] [Q&A] [answered]
阿里网盘里的图片文件，用网盘客户端下载下来是正常的，可是用webdav就不正常，打开显示文件被截断，缩略图预览底下一部分失效了。
右键查看文件属性是一样的，文件大小和占用空间**一模一样**。但md5是**不一样**的，通过ultraEdit查看二进制可以看到文件除了文件头部分，后面的数据也是**完全不一样**的，利用beyondCompare对比也是一样的结论。
以下是图片文件的一些对比：
![对比](https://user-images.githubusercontent.com/13639590/161569266-be0a9f3d-7c95-403e-a335-649058d643a4.jpg)
以下分别是原图和webdav里的图：
![原文件](https://user-images.githubusercontent.com/13639590/161565143-f66ece92-4a40-499b-a097-b4669f2097be.jpg)
![webdav](https://user-images.githubusercontent.com/13639590/161565094-ec2d8160-172f-4696-8577-3461759bf1a9.jpg)

另外我又对比了一些其他格式的文件，发现只有图片格式的文件数据会被修改。
虽然不少图片文件的数据被修改后仍然显示正常，但也有不小的一部分图就坏了（通过网盘网页版和客户端查看下载都是正常的，只有webdav有问题）
![对比2](https://user-images.githubusercontent.com/13639590/161571782-f39998ef-b3c7-465d-bea2-925fdfdf49b3.jpg)

因此想来问问看，是会对图片文件进行数据修改吗？如果是为了压缩大小我还能理解，但是文件大小完全没变，只是把内容改坏了我不是很理解....

**Comment by messense** [comment, marked as answer]
除了 .livp 格式的文件需要重新组合，其他文件都是阿里云盘提供什么下载什么，WebDAV 协议没有 checksum 可能是下载的文件传输过程中损坏了。

**Reply by zhendery** [reply]
> 除了 .livp 格式的文件需要重新组合，其他文件都是阿里云盘提供什么下载什么，WebDAV 协议没有 checksum 可能是下载的文件传输过程中损坏了。

很怪，我测试了jpg和png格式的文件都会这样，虽然有的图片被修改后显示正常。
另外我用了另一个webdav的项目测试了也是这样，我怀疑是webdav本身的问题，或者是阿里云盘的问题。
那个项目：https://github.com/tickstep/aliyunpan/issues/78

**Reply by westhiker** [reply]
@zhendery 我测试了下你发的这个[项目](https://github.com/tickstep/aliyunpan) ，图片是可以正常下载，无修改的...

**Reply by westhiker** [reply]
如果是下载过程中损坏，应该是随机的，不应该每次访问得到的残缺图片都是一样的。

**Reply by zhendery** [reply]
> @zhendery 我测试了下你发的这个[项目](https://github.com/tickstep/aliyunpan) ，图片是可以正常下载，无修改的...

真的欸，那个项目我换了个端口就好了，那么看来只有你的项目有这个问题哈哈哈（笑哭）

**Reply by westhiker** [reply]
是的，目前我先退回了1.3.0，没有这个问题。

**Reply by zhendery** [reply]
> 是的，目前我先退回了1.3.0，没有这个问题。

确实，我也退回1.3.0了，一切OK

**Reply by messense** [reply]
1.3.1 的改动只是复用了阿里云盘列表接口的下载文件 url 来降低获取文件下载 url 接口的调用次数降低被阿里云盘 429 限制的可能性，如果 1.3.0 正常的话那可能是阿里云盘列表接口给的下载文件 url 有些问题。

**Reply by messense** [reply]
试试 1.3.2

**Reply by sailing178** [reply]
> 是的，目前我先退回了1.3.0，没有这个问题。
![image](https://user-images.githubusercontent.com/89731041/175085730-09c693d4-8899-4835-accf-17dd669d2b38.png)

我想问下为什么我退货1.30还是有问题呢

**Comment by LIUBANGJIAN** [comment]
请问，这个问题怎么解决呢？

**Reply by LIUBANGJIAN** [reply]
1.3.2有修复计划吗

**Reply by LIUBANGJIAN** [reply]
我用edge浏览器，看jellyfin海报是正常的，但是ios的infuse播放器看海报又是少了半截。

**Reply by zhendery** [reply]
暂时用回1.3.0吧，作者说的1.3.1的改动感觉也不是特别必要，只要不非常频繁调用还是可以避免的。
然后我感觉这个问题不是作者能控制的，得要阿里云盘那边把接口改对了才行。

**Reply by messense** [reply]
试试 1.3.2

**Reply by sailing178** [reply]
同问题，真的晕了

**Comment by sailing178** [comment]
退回127已正常

**Reply by sailing178** [reply]
第二天实测，127如出一辙
![image](https://user-images.githubusercontent.com/89731041/175209206-af86ef3d-1e2b-460b-b673-da251ef3bc71.png)

**Comment by Gokou-Ruri** [comment]
完全不存在这个问题，应该是您的网络丢包太严重了。
我测试了几百张图片都没有出现下载损坏。
而且你这也弄得太复杂了，您最上面第一张截图明显就是图片下载过程中出现了数据损坏，根本就不是什么数据被修改，如果您认为是数据修改，那么是修改了什么数据？您认为修改什么数据会导致图片丢一半内容？
![image](https://user-images.githubusercontent.com/26711804/177003706-6796814b-f9bf-438b-a5b5-96c5df641686.png)

**Reply by zhendery** [reply]
？？？ 作者自己都回复说确实有这个问题并给出使用建议了，你测试的版本和我不一样吧
